extensions/prometheus/PublishedMetricGaugeCollection.cpp

```diff
@@ -0,0 +1,49 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "PublishedMetricGaugeCollection.h"
+
+#include <utility>
+#include <algorithm>
+
+#include "prometheus/client_metric.h"
+#include "state/PublishedMetricProvider.h"
+#include "range/v3/range/conversion.hpp"
+#include "range/v3/view/transform.hpp"
+
+namespace org::apache::nifi::minifi::extensions::prometheus {
+
+PublishedMetricGaugeCollection::PublishedMetricGaugeCollection(std::shared_ptr<state::PublishedMetricProvider> metric) : metric_{std::move(metric)} {
+}
+
+std::vector<::prometheus::MetricFamily> PublishedMetricGaugeCollection::Collect() const {
+  std::vector<::prometheus::MetricFamily> collection;
+  for (const auto& metric : metric_->calculateMetrics()) {
+    ::prometheus::ClientMetric client_metric;
+    client_metric.label = ranges::views::transform(metric.labels, [](auto&& kvp) { return ::prometheus::ClientMetric::Label{kvp.first, kvp.second}; })
+      | ranges::to<std::vector<::prometheus::ClientMetric::Label>>;
+    client_metric.gauge = ::prometheus::ClientMetric::Gauge{metric.value};
+    collection.push_back({
+      .name = metric.name,
+      .help = "",
+      .type = ::prometheus::MetricType::Gauge,
+      .metric = { client_metric }
+    });
```

Comment:
I think it would help if metric names were prefixed with "minifi_" before serving the data to prometheus. It just makes finding them much easier if multiple programs with different metrics are used.

![image](https://user-images.githubusercontent.com/1170582/173875908-d120e0f3-4aeb-4da9-a478-a1442ffebebc.png)

Reply:
Most guidelines discouraged prefixes for metric names and I didn't want to use it because Prometheus already adds a `job` label for the metric with the `minifi` value that can be filtered for. But I checked the [Prometheus best practices](https://prometheus.io/docs/practices/naming/) and it says that the metrics should have an application name as a prefix to mark the domain (or sometimes it's not an application name but something general like `http`). In our case using `minifi_` prefix is a good idea I added it in 96431d07a867a26723d9d6c2cf71edc1d1dd0b17